diff --git a/lib/features/find_friends/screens/find_friends_screen.dart b/lib/features/find_friends/screens/find_friends_screen.dart
index f7fef0aaa7ba735b4bccfeb25b2f5052d6f64661..0ee141d51cd1de1b85dbf165e846da8e75d55166 100644
--- a/lib/features/find_friends/screens/find_friends_screen.dart
+++ b/lib/features/find_friends/screens/find_friends_screen.dart
@@ -1,10 +1,16 @@
 import 'package:flutter/material.dart';
 import '../../../core/models/user_model.dart';
 
 class FindFriendsScreen extends StatelessWidget {
-  final UserModel? userModel;
-  const FindFriendsScreen({this.userModel, super.key});
+  /// Existing user profile for the current user.
+  final UserModel? profile;
+
+  /// Callback triggered when the profile is saved or updated.
+  final VoidCallback? onSave;
+
+  const FindFriendsScreen({this.profile, this.onSave, super.key});
+
   @override
   Widget build(BuildContext context) =>
       const Center(child: Text('Find Friends Screen'));
 }
diff --git a/lib/features/main_screen/home_screen.dart b/lib/features/main_screen/home_screen.dart
index f238c0929ce8a62bc4c030fb5035fb19f99c9ed6..018443923442339afd216cf61b7913239376fa07 100644
--- a/lib/features/main_screen/home_screen.dart
+++ b/lib/features/main_screen/home_screen.dart
@@ -246,51 +246,51 @@ class _HomeScreenState extends State<HomeScreen> with TickerProviderStateMixin {
                   child: Row(
                     mainAxisAlignment: MainAxisAlignment.center,
                     children: [
                       Icon(tab['icon']),
                       const SizedBox(width: 8),
                       Text(tab['key'].toString().tr()),
                     ],
                   ),
                 ))
             .toList(),
       ),
     );
   }
 
   Widget _buildHomePage() {
     return NestedScrollView(
       headerSliverBuilder: (context, innerBoxIsScrolled) => [
             _buildHomeSliverAppBar(),
           ],
       body: TabBarView(
         controller: _tabController,
         children: [
           MainFeedScreen(userModel: _userModel),
           LocalNewsScreen(userModel: _userModel),
           MarketplaceScreen(userModel: _userModel),
-          FindFriendsScreen(userModel: _userModel),
+          FindFriendsScreen(profile: _userModel),
           ClubsScreen(userModel: _userModel),
           JobsScreen(userModel: _userModel),
           LocalStoresScreen(userModel: _userModel),
           AuctionScreen(userModel: _userModel),
           PomScreen(userModel: _userModel),
         ],
       ),
     );
   }
 
   @override
   Widget build(BuildContext context) {
     final List<Widget> pages = [
       _buildHomePage(),
       const SearchScreen(),
       const ChatListScreen(),
       const MyBlingScreen(),
     ];
 
     int effectiveIndex =
         _bottomNavIndex > 2 ? _bottomNavIndex - 1 : _bottomNavIndex;
 
     return Scaffold(
       appBar: null,
       drawer: _buildAppDrawer(_userModel),
